src/wp-includes/blocks/footnotes.php

```diff
@@ -78,3 +82,137 @@ function register_block_core_footnotes() {
 	);
 }
 add_action( 'init', 'register_block_core_footnotes' );
+
+add_action(
+	'wp_after_insert_post',
+	/**
+	 * Saves the footnotes meta value to the revision.
+	 *
+	 * @since 6.3.0
+	 *
+	 * @param int $revision_id The revision ID.
+	 */
+	static function( $revision_id ) {
+		$post_id = wp_is_post_revision( $revision_id );
+
+		if ( $post_id ) {
+			$footnotes = get_post_meta( $post_id, 'footnotes', true );
+
+			if ( $footnotes ) {
+				// Can't use update_post_meta() because it doesn't allow revisions.
+				update_metadata( 'post', $revision_id, 'footnotes', $footnotes );
+			}
+		}
+	}
+);
+
+add_action(
+	'_wp_put_post_revision',
+	/**
+	 * Keeps track of the revision ID for "rest_after_insert_{$post_type}".
+	 *
```

Comment:
It's missing a since declaration here :)

Reply:
Ah, so anonymous functions should also receive `@since`?

Reply:
Why are these all anonymous functions in the first place?

Makes it impossible for developers to unhook them.

Reply:
I personally like anonymous functions for handling the footnote meta because it makes the way to move to handling postmeta revisions [in a more standard way in 6.4](https://core.trac.wordpress.org/ticket/20564).

Unless you think we'd still want the named functions after doing so?

Reply:
That would indeed be an argument for keeping them, but not sure if that's the reason for it?

Reply:
Because this is a temporary fix and to avoid having to rename/naming clashes

Reply:
Does this seem like an appropriate reason for using anonymous functions to you, @swissspidy ? (+cc: @peterwilsoncc and @joemcgill ).

If not, do you think we should continue discussion about this in the [existing issue about revision support](https://github.com/WordPress/gutenberg/issues/52541) or open a new issue?

It sounded like there may be platform considerations for using named functions here, and I want to make sure those are well understood.

Reply:
It is, yes